dotCMS/src/main/java/com/dotcms/storage/ContentletMetadata.java

```diff
@@ -0,0 +1,27 @@
+package com.dotcms.storage;
+
+import java.util.Map;
+
+/**
+ * Encapsulates a collection with field with full metadata generation and
+ * all fields with basic metadata
+ */
+public class ContentletMetadata {
```

Comment:
This is just a class that presents them both as (as an aggregation I suppose) since Full metadata isn't meant to live in cache. 

The metadata maps are already filtered/cached

Reply:
Only basic md lives in cache

Reply:
Modified the type of the maps to be Map<String, Serialiable> to ensure they'll make it safe into cache